godtools/Managers/FirstLaunchInitializer.swift

```diff
@@ -15,84 +15,85 @@ class FirstLaunchInitializer: GTDataManager {
         let predicate = NSPredicate(format: "remoteId = %@", magicId)
         
-        deleteEntities(Translation.self, matching: predicate)
-        deleteEntities(Language.self, matching: predicate)
-        deleteEntities(DownloadedResource.self, matching: predicate)
-        
-        if GTSettings.shared.primaryLanguageId == magicId {
-            let primaryLanguageCode = Locale.preferredLanguages.first ?? "en"
-            let primaryLanguage = findEntity(Language.self, byAttribute: "code", withValue: primaryLanguageCode) ??
-                findEntity(Language.self, byAttribute: "code", withValue: "en")
-        
-            primaryLanguage?.shouldDownload = true
+        try! realm.write {
+            realm.delete(findEntities(Translation.self, matching: predicate))
+            realm.delete(findEntities(Language.self, matching: predicate))
+            realm.delete(findEntities(DownloadedResource.self, matching: predicate))
             
-            for resource in findEntities(DownloadedResource.self, matching: NSPredicate(format:"code IN %@", initialPackageCodes)) {
-                resource.shouldDownload = true
+            if GTSettings.shared.primaryLanguageId == magicId {
+                let primaryLanguageCode = Locale.preferredLanguages.first ?? "en"
+                let primaryLanguage = findEntity(Language.self, byAttribute: "code", withValue: primaryLanguageCode) ??
+                    findEntity(Language.self, byAttribute: "code", withValue: "en")
+                
+                primaryLanguage?.shouldDownload = true
+                
+                for resource in findEntities(DownloadedResource.self, matching: NSPredicate(format:"code IN %@", initialPackageCodes)) {
+                    resource.shouldDownload = true
+                }
+                
+                GTSettings.shared.primaryLanguageId = primaryLanguage?.remoteId
             }
             
-            GTSettings.shared.primaryLanguageId = primaryLanguage?.remoteId
         }
-        
-        saveToDiskAndWait()
-        
         sendCompletedNotification()
     }
     
-    private func initializeInitialLanguage() -> Language? {
-        let language = createEntity(Language.self)
+    private func initializeInitialLanguage() -> Language {
+        let language = Language()
         
-        language?.code = "en-US"
-        language?.remoteId = magicId
+        language.code = "en-US"
+        language.remoteId = magicId
         
         return language
     }
     
-    private func initializeInitialResources(language: Language?) -> [DownloadedResource?] {
-        let kgp = createEntity(DownloadedResource.self)
-        kgp?.code = "kgp"
-        kgp?.name = "Knowing God Personally"
-        kgp?.remoteId = magicId
-        kgp?.shouldDownload = true
+    private func initializeInitialResources(language: Language?) -> [DownloadedResource] {
+        let kgp = DownloadedResource()
+        kgp.code = "kgp"
```

Comment:
Correct. We want to pre-load the app with some options so that the users can still run the app if they cannot connect to the API. So these values have to be hard-coded, but there may be a better way to do it.